### PR TITLE
Implement dirty state

### PR DESCRIPTION
**Pull request by b3n4kh**
Fixes: #342 

Implement a dirty state feature in the ProductView component to track unsaved changes and alert users with a warning icon on the Save button.

New Features:
- Introduce a dirty state indicator in the ProductView component to alert users of unsaved changes.

Enhancements:
- Add a Save button with a warning icon in the ProductView component to indicate unsaved changes.

## Summary by Sourcery

Implement a dirty state feature in the ProductView component to track unsaved changes, alert users with a warning icon on the Save button, and provide a dialog for handling unsaved changes.

New Features:
- Introduce a dirty state feature in the ProductView component to track unsaved changes and alert users with a warning icon on the Save button.

Enhancements:
- Add a Save button with a warning icon in the ProductView component to indicate unsaved changes.

### CODE REVIEW

**Comment by stefaniasava** [comment]
### **Introduction of Dirty State Management for Product Editor**

This PR introduces a dirty state system to the product editor, ensuring users are alerted to unsaved changes before taking actions that could lead to data loss. This addition brings a more robust and user-friendly experience when working with product data.

**Changes**:

- Implemented a dirty state that monitors changes to the product form. Any modifications to fields like title, description, and product type now toggle the dirty state to true, visually indicated by the Save button turning orange.
- Added a PopupDirty dialog component that appears when users attempt to publish or render a product with unsaved changes.
- Introduced attemptPublish() and attemptRerenderProduct() functions. These functions check the dirty state and trigger the confirmation dialog if there are unsaved changes, ensuring actions only proceed after the user confirms.
- Updated the ctrl+p hotkey for publishing to use attemptPublish(), aligning the keyboard shortcut with the dirty state handling. If unsaved changes are detected, the hotkey now prompts users with the confirmation dialog instead of directly publishing.

New Features:
- Introduce a dirty state indicator in the ProductView component to alert users of unsaved changes.

Enhancements:
- Add a Save button with a warning icon in the ProductView component to indicate unsaved changes.

File: src/gui/src/components/popups/PopupDirty.vue [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      this.$emit('cancel');
+      this.internalModelValue = false // Close the dialog
+    },
+    onSaveAndContinue() {
+      this.$emit('saveAndContinue');
+      this.internalModelValue = false // Close the dialog after saving
+    },
+  },
+}
+</script>
```
Comment (b3n4kh) [inline src/gui/src/components/popups/PopupDirty.vue:77]: Use composition API: https://vuejs.org/api/composition-api-setup

File: src/gui/src/components/popups/PopupDirty.vue [hunk 2]
```diff
@@ -0,0 +1,96 @@
+<template>
+  <v-dialog v-model="internalModelValue" width="400">
```
Comment (b3n4kh) [inline src/gui/src/components/popups/PopupDirty.vue:2]: use `modelValue` directly

**Comment by stefaniasava** [comment]
Note: The e2e tests are currently failing due to a missing pnpm dependency in the CI environment. A fix for this issue has been implemented in PR #359 , which updates the CI workflowso that pnpm is properly installed and configured. Once that PR is merged, the e2e tests in this PR should pass without issues.

File: src/gui/src/components/popups/PopupDirty.vue [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  name: 'PopupDirty',
+  props: {
+    modelValue: {
+      type: Boolean,
+      required: true,
+    },
+  },
+  emits: ['update:modelValue', 'saveAndContinue', 'cancel'],
+  setup(props, { emit }) {
+    const dialogValue = ref(props.modelValue)
```
Comment (b3n4kh) [inline src/gui/src/components/popups/PopupDirty.vue:55]: Is now obsolete after the switch to directly using `modelValue` and can be removed.

**Pull request merged**